circuitbreaker/hystrix.go

```diff
@@ -0,0 +1,55 @@
+package circuitbreaker
+
+import (
+	"errors"
+
+	"github.com/afex/hystrix-go/hystrix"
+	"github.com/go-kit/kit/endpoint"
+	"golang.org/x/net/context"
+)
+
+var ErrCircuitBreakerTimeout = errors.New("circuit breaker timeout")
+
+// Hystrix returns an endpoint.Middleware that implements the circuit
+// breaker pattern using the afex/hystrix-go package.
+//
+// When using this circuit breaker, please configure your commands separately.
+//
+// See https://godoc.org/github.com/afex/hystrix-go/hystrix for more
+// information.
+func Hystrix(commandName string) endpoint.Middleware {
+	return func(next endpoint.Endpoint) endpoint.Endpoint {
+		return func(ctx context.Context, request interface{}) (response interface{}, err error) {
+			output := make(chan interface{}, 1)
+			errors := hystrix.Go(commandName, func() error {
+				resp, err := next(ctx, request)
+				if err == nil {
+					output <- resp
+				}
+				return err
+			}, nil)
+
+			return selectOutput(output, errors)
+		}
+	}
+}
+
+func selectOutput(output <-chan interface{}, errors <-chan error) (interface{}, error) {
+	select {
+	case resp := <-output:
+		return resp, nil
+	case err := <-errors:
+		return nil, wellKnownHystrixError(err)
+	}
+}
+
+func wellKnownHystrixError(err error) error {
+	switch err {
+	case hystrix.ErrCircuitOpen:
+		return ErrCircuitBreakerOpen
+	case hystrix.ErrTimeout:
+		return ErrCircuitBreakerTimeout
+	default:
+		return err
+	}
+}
```

Comment:
There may indeed be value in doing that, but I think I prefer the idea of defaulting to transparency, and just returning the errors directly. If it turns out users want consistent errors independent of concrete implementation, we can add it in by request, later.